### PR TITLE
Simplify Travis

### PR DESCRIPTION
**Pull request by elazarg**

### CODE REVIEW

**Comment by elazarg** [comment]
@pchaigno I hope these changes work for you (particularly the docker file). Please let me know if there's any issue.

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-RUN pip3 install matplotlib
+RUN apt-get -yq --no-install-suggests --no-install-recommends install build-essential cmake libgmp-dev libboost-dev
 
 WORKDIR /verifier
 COPY . /verifier/
 RUN mkdir build
 WORKDIR /verifier/build
 RUN cmake .. -DCMAKE_BUILD_TYPE=Release
-RUN make -j
+RUN make -j4
```
Comment (pchaigno) [inline Dockerfile]: Same here. Why hardcode 4?

File: .travis.yml [hunk 2]
```diff
@@ -2,26 +2,17 @@ language: cpp
...
-        - CXX: g++-8
-      before_install:
-        - sudo apt update
-        - sudo apt install cmake libboost-dev
+            - libboost-dev
       install:
         - mkdir build
         - cd build
         - cmake .. -DCMAKE_BUILD_TYPE=Release
+        - make -j3
```
[the remark this answers is not in the document]
Comment (elazarg) [reply .travis.yml]: AFAIK, `-j` is not generic, but spawns all possible processes at once, which is far from optimal. Also, it does not give any hint about the progress - you have to just wait for the linking.

[the remark this answers is not in the document]
Comment (pchaigno) [reply .travis.yml]: Oh, I just learned something :-)

What about `make -j$(nproc)`, then?

[the remark this answers is not in the document]
Comment (elazarg) [reply .travis.yml]: makes sense, though specifically in Travis it doesn't work well.

[the remark this answers is not in the document]
Comment (pchaigno) [reply .travis.yml]: Arf! What's happening? `nproc` returns 2 for me in Travis CI.

It's more important to me in the Dockerfile anyway. I don't want to build using 4 cores when I have 16 available.

[the remark this answers is not in the document]
Comment (elazarg) [reply .travis.yml]: I pushed `make -j $(nproc)` in Dockerfile directly to master

[the remark this answers is not in the document]
Comment (elazarg) [reply .travis.yml]: > Arf! What's happening? `nproc` returns 2 for me in Travis CI.
> 
I'm not really sure, but the internet told me not to trust it. Maybe the internet is wrong. 🤷‍♂

[the remark this answers is not in the document]
Comment (pchaigno) [reply .travis.yml]: I think it used to return the number of cores on the host when running inside a container. It looks fixed now.

**Pull request merged**